### PR TITLE
Old way of bz checker initialization

### PR DESCRIPTION
**Pull request by jstavel**
There is a class OldBzChecker.
The class offers a static method to get instance of BzChecker.
If you do not use google injection you can use this static factory class.

I have tested usage of this class in jenkins environment. With success.

### CODE REVIEW

**Comment by Lorquas** [comment]
@jstavel what changes do I need to make to the automation.properties (or otherwise) for this to work?

**Comment by jstavel** [comment]
Hello,
it is not necessary to change anything in case old xmlrpc API is used.
If you want to use new REST-API, you copy/paste a key (from bugzilla web admin) into a new variable 'bugzilla.apikey'. I have added this key into the file 'automation.properties' that is in our code base.

File: src/rhsm/cli/tasks/CandlepinTasks.java [hunk 1]
```diff
@@ -699,7 +699,7 @@ static public String getResourceUsingRESTfulAPI(String authenticator, String pas
...
 				// duplicate of Bug 1113741 - RHEL 7 (and 6?): subscription-manager fails with "JSON parsing error: No JSON object could be decoded" error
-				try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (XmlRpcException xre) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}
+				try {if (invokeWorkaroundWhileBugIsOpen&&OldBzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+OldBzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (BugzillaAPIException be) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}  catch (Exception ex) {/* ignore exception */}
```
Comment (jsefler) [inline src/rhsm/cli/tasks/CandlepinTasks.java]: Why are we appending this catch all Exceptions at the end of these try catch lines?
  catch (Exception ex) {/* ignore exception */}

I think we should NOT catch Exception ex

This applies to A LOT of line changes in this pull request

**Comment by jsefler** [comment]
Why are we appending this catch all Exceptions at the end of all the WORKAROUND try catch lines?
      catch (Exception ex) {/* ignore exception */}

I think we should NOT catch Exception ex

This applies to a lot of line changes in this pull request

File: src/rhsm/cli/tasks/CandlepinTasks.java (hunk 1 above)
Comment (jstavel) [inline src/rhsm/cli/tasks/CandlepinTasks.java]: You're right. The catch is redundand in this line. I'll remove it.

Comment (jstavel) [inline src/rhsm/cli/tasks/CandlepinTasks.java]: I have removed the code `catch Exception`.  I mean all lines where bugzilla is used. There are a few `catch Exception` pieces with different context in the testware. I have left those pieces of a code.

**Comment by jstavel** [comment]
I will:
- join 'OldBzChecker' with 'BzChecker'
- add dispatching of bugzilla implementation based on `bugzilla.api.type` (REST, XMLRPC)

**Comment by jstavel** [comment]
Those PRs come to play
https://github.com/RedHatQE/bz-checker/pull/1
https://github.com/RedHatQE/bugzilla-testng/pull/1

**Comment by jstavel** [comment]
All remarks were implemented. I will start functional testing of that pile.

File: src/rhsm/base/SubscriptionManagerCLITestScript.java [hunk 2]
```diff
@@ -593,7 +593,7 @@ public void ensureSELinuxIsEnforcingBeforeSuite() {
 				// TEMPORARY WORKAROUND
 				invokeWorkaroundWhileBugIsOpen = true;
 				bugId="1342401";	// Bug 1342401 - Allow NetworkManager to create temporary /etc/resolv.conf.XXXXXX file 
-				try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (XmlRpcException xre) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}
+				try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (BugzillaAPIException be) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}  catch (Exception e) {/* ignore exception */}
```
Comment (jsefler) [inline src/rhsm/base/SubscriptionManagerCLITestScript.java]: remove the last  catch (Exception e) {/* ignore exception */}

File: src/rhsm/base/SubscriptionManagerCLITestScript.java [hunk 3]
```diff
@@ -602,7 +602,7 @@ public void ensureSELinuxIsEnforcingBeforeSuite() {
...
 				invokeWorkaroundWhileBugIsOpen = true;
 				bugId="1343648";	// Bug 1343648 - SELinux label for /etc/udev/hwdb.bin is etc_t instead of systemd_hwdb_etc_t after "#systemd-hwdb update" 
-				try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (XmlRpcException xre) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}
+				try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (BugzillaAPIException be) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}  catch (Exception e) {/* ignore exception */}
```
Comment (jsefler) [inline src/rhsm/base/SubscriptionManagerCLITestScript.java]: remove the last  catch (Exception e) {/* ignore exception */}

File: src/rhsm/base/SubscriptionManagerCLITestScript.java [hunk 4]
```diff
@@ -611,7 +611,7 @@ public void ensureSELinuxIsEnforcingBeforeSuite() {
...
 				invokeWorkaroundWhileBugIsOpen = true;
 				bugId="1350756";	// Bug 1350756 - SELinux label for /etc/udev/hwdb.bin is etc_t instead of systemd_hwdb_etc_t after "#systemd-hwdb update" 
-				try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (XmlRpcException xre) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}
+				try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (BugzillaAPIException be) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}  catch (Exception e) {/* ignore exception */}
```
Comment (jsefler) [inline src/rhsm/base/SubscriptionManagerCLITestScript.java]: remove the last  catch (Exception e) {/* ignore exception */}

File: src/rhsm/base/SubscriptionManagerCLITestScript.java [hunk 5]
```diff
@@ -647,7 +647,7 @@ public void verifyNoSELinuxDenialsWereLoggedAfterClass() {
...
 					String bugId="1362273"; // Bug 1362273 - avc denied /var/log/audit/audit.log when "systemd: Started Session # of user root." is written to /var/log/messages every two minutes
-					try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (XmlRpcException xre) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}
+					try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (BugzillaAPIException be) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}  catch (Exception e) {/* ignore exception */}
```
Comment (jsefler) [inline src/rhsm/base/SubscriptionManagerCLITestScript.java]: remove the last  catch (Exception e) {/* ignore exception */}

File: src/rhsm/base/SubscriptionManagerCLITestScript.java [hunk 6]
```diff
@@ -2063,7 +2063,7 @@ protected void verifyCurrentEntitlementCertsAreReflectedInCurrentlySubscribedYum
 			if (username.equals("anonymous")) {
 				boolean invokeWorkaroundWhileBugIsOpen = true;
 				String bugId="741961"; 
-				try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (XmlRpcException xre) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */}
+				try {if (invokeWorkaroundWhileBugIsOpen&&BzChecker.getInstance().isBugOpen(bugId)) {log.fine("Invoking workaround for "+BzChecker.getInstance().getBugState(bugId).toString()+" Bugzilla "+bugId+".  (https://bugzilla.redhat.com/show_bug.cgi?id="+bugId+")");SubscriptionManagerCLITestScript.addInvokedWorkaround(bugId);} else {invokeWorkaroundWhileBugIsOpen=false;}} catch (BugzillaAPIException be) {/* ignore exception */} catch (RuntimeException re) {/* ignore exception */} catch (Exception e) {/* ignore exception */}
```
Comment (jsefler) [inline src/rhsm/base/SubscriptionManagerCLITestScript.java]: remove the last  catch (Exception e) {/* ignore exception */}

Comment (jstavel) [inline src/rhsm/base/SubscriptionManagerCLITestScript.java]: Removed `catch Exception`.

**Pull request merged**